### DISCUSSION TITLE
lamda with nodemailer

### DISCUSSION THREAD

**Discussion by pakman198** [vercel/vercel#4496] [Help] [answered]
Hello

This is my first time creating a serverless app.
I created an api endpoint that when is reached, will generate an email and send it via [nodemailer](https://nodemailer.com/about/#example).

While working on the development, i tested my endpoint and was able to receive emails, but when I deployed the app, nodmailer stays hanging in 
`DEBUG Sending mail using SMTP/6.4.6[client:6.4.6]`

I already logged the env variables and my credentials, host and port are correct, so I don't know if I might need to do something else.

Any help is appreciated :)

**Comment by mcsdevv** [comment, marked as answer]
Hello @pakman198! It is not possible to use SMTP with Vercel, as it blocked by an upstream provider. I'd recommend using a third-party service to handle this.

Further information available here:

- [Next.js + SendGrid guide](https://vercel.com/guides/deploying-nextjs-nodejs-and-sendgrid-with-vercel)
- [Vercel SMTP Limitation](https://vercel.com/docs/v2/platform/limits#smtp)

**Reply by pakman198** [reply]
Thank you !

**Reply by ullaskunder3** [reply]
ooook.... I had two different repo and using vercel and nodemailer for email confirmation it works on development and not working in production...😭I was going nuts over it... now I get it

**Comment by ullaskunder3** [comment]
is there any demonstrated solution or any examples ?

**Reply by amyegan** [reply]
Thanks for reaching out @ullaskunder3! The original post is a couple of years old and it looks like the original answer might be a bit out of date. We have some docs about  [using SMTP with Vercel](https://vercel.com/support/articles/serverless-functions-and-smtp) and [email with Vercel](https://vercel.com/docs/concepts/solutions/email) that might help.

  > Sending emails from within your application can be achieved by using [Serverless Functions](https://vercel.com/docs/concepts/functions). To do this, you can either rely on the outgoing SMTP connection of your Serverless Function or use the API of an external service.

  > You can also use libraries like [nodemailer](https://www.npmjs.com/package/nodemailer). When doing so, we recommend using an external host value provided by a third-party provider, rather than defaulting to the outbound SMTP connection of the Serverless Function.

I don't believe we have a full example app to demonstrate this, but that is a great idea! I'll add it to my suggestions list of things to build. 

If you need more information or assistance, please start a new discussion at https://github.com/orgs/vercel/discussions

**Comment by ullaskunder3** [comment]
@amyegan Done... I fixed my problem [ I cried when it worked 😅 ]
- I switched from nodemailer to sendGrid (besttttttt option) DONEEEEE

```
const sgMail = require('@sendgrid/mail')
sgMail.setApiKey(process.env.SENDGRID_API_KEY);

const emailTemplate = {
    verifyUserUrl: process.env.VERIFY_USER_URL
}

module.exports = async (email, verifyUserUrl) => {
    const mailData = {
        to: email,
        from: process.env.ADMINEID,
        subject: 'Welcome to nxt devloper Hub.',
        templateId: emailTemplate.verifyUserUrl,
        dynamic_template_data:{
            email: email,
            verifyUserUrl: verifyUserUrl
        }

    }
    try {
        await sgMail.send(mailData);
    } catch (error) {
        if(error.response){
            console.log(':ERR body:', error.response.body);
        }
    }
}
```

**Reply by pakman198** [reply]
I also ended up using SendGrid.
Amazing solution and super easy to implement.
